validator/testcases/regex.py

```diff
@@ -2901,9 +2915,34 @@ def munge_filename(name):
                         "the `%s` preference branch" % branch)))
         for branch, reason in BANNED_PREF_BRANCHES))
 
-STRING_REGEXPS += PREF_REGEXPS
+# For tests in literal strings, add help text suggesting passing the
+# preference directly to reference getter functions.
+PREF_STRING_HELP = (
+    "If you are reading, but not writing, this preference, please consider "
+    "passing a string literal directly to `Preferences.get()` or "
+    "`nsIPrefBranch.get*Pref`.")
+
+
+def maybe_tuple(value):
+    if isinstance(value, tuple):
+        return value
+    return (value,)
+
+
+def add_pref_help(desc):
+    desc = desc.copy()
+    for key in "description", "signing_help":
+        if key in desc:
+            desc[key] = maybe_tuple(desc[key]) + maybe_tuple(PREF_STRING_HELP)
```

Comment:
If `description` or `signing_help` isn't in the dictionary then this won't add `PREF_STRING_HELP` on its own, is that intended?